optima/programs.py

```diff
@@ -578,10 +600,35 @@ def compareoutcomes(self, parset=None, year=None, ind=0, doprint=False):
     
     
     
-    def reconcilewithpars(self, parset=None, year=None, ind=0):
+    def cco2odict(self, t=None):
+        ''' Parse the cost-coverage-outcome tree and pull out parameter values into an odict '''
+        if t is None: raise Exception('Please supply a year')
+        modifiablepars = odict()
+        for targetpartype in self.covout.keys():
+            for targetparpop in self.covout[targetpartype].keys():
+                modifiablepars[(targetpartype,targetparpop,'intercept')] = [self.covout[targetpartype][targetparpop].getccopar(t=t,bounds='l')['intercept'][0],self.covout[targetpartype][targetparpop].getccopar(t=t,bounds='u')['intercept'][0]]
+                for thisprog in self.progs_by_targetpar(targetpartype)[targetparpop]:
+                    try: modifiablepars[(targetpartype,targetparpop,thisprog.short)] = [self.covout[targetpartype][targetparpop].getccopar(t=t,bounds='lower')[thisprog.short][0], self.covout[targetpartype][targetparpop].getccopar(t=t,bounds='upper')[thisprog.short][0]]
+                    except: pass # Must be something like ART, which does not have adjustable parameters -- WARNING, could test explicitly!
```

Comment:
this is going to be much easier to handles later on, when i change programs to use Par objects directly.